File: Week04/LeetCode-0217.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        """
        O(N) algorithm with O(N) extra space
        """
        mySet = set()
        for num in nums:
            if num in mySet:
                return True
            else:
                mySet.add(num)        
        return False

File: Week04/LeetCode-0217.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        // O(N) algorithm with O(N) extra space
        unordered_set<int> mySet;
        for (auto num: nums){
            if (mySet.count(num) != 0)
                return true;
            else
                mySet.insert(num);
        } // end-for
        
        return false;        
    } // end-containsDuplicate
};